rules/no-empty-file.js

```diff
@@ -5,13 +5,24 @@ const messages = {
 	[MESSAGE_ID]: 'Empty files are not allowed.',
 };
 
+const isTripleSlashDirective = node =>
+	node.type === 'Line' && node.value.startsWith('/');
+
 const isEmpty = node =>
 	(
-		(node.type === 'Program' || node.type === 'BlockStatement')
+		node.type === 'Program'
+		&& node.body.every(currentNode => isEmpty(currentNode))
+		&& node.comments.every(currentNode => !isTripleSlashDirective(currentNode))
```

Comment:
I don't think we can use `node.comment`. See https://eslint.org/docs/developer-guide/working-with-rules#sourcecodegetallcomments

I also found [shebang is stored as `Shebang`](https://eslint.org/docs/developer-guide/working-with-rules#accessing-shebangs).

Reply:
Yep! `comments` are `Line`, `Block`, and `Shebang`. I was messing with that before I simplified. 

The comments are showing up in `node.comments` as I'd expect if you debug or `console.log`, but I can look into switching it over to use `sourceCode.getAllComments()` instead, if you prefer.

Reply:
@fisker PR updated again.

Reply:
> The comments are showing up in node.comments 

It's not safe for other nodes except `Program`.  

> For example, the @babel/eslint-parser can set attachComment to false since the linter does not care comment attachment. We even delete such comments in @babel/eslint-parser.

Quote from https://github.com/babel/babel/pull/13229#issue-871065828